cedar-policy/src/api.rs

```diff
@@ -3633,6 +3643,36 @@ impl Context {
         )?))
     }
 
+    /// Retrieves a reference to a Value from the Context by its key.
+    ///
+    /// # Arguments
+    ///
+    /// * `key` - The key to look up in the context
+    ///
+    /// # Returns
+    ///
+    /// * `Some(&Value)` - If the key exists in the context, returns a reference to its value
+    /// * `None` - If the key doesn't exist or if the context is not a Value type
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use cedar_policy::{Context, Request, EntityUid};
+    /// # use std::str::FromStr;
+    /// let context = Context::from_json_str(r#"{"rayId": "abc123"}"#, None).unwrap();
+    /// if let Some(value) = context.get("rayId") {
+    ///     // value here is a &Value, not necessarily a string
+    ///     println!("Found value: {:?}", value);
+    /// }
+    /// assert_eq!(context.get("nonexistent"), None);
+    /// ```
+    pub fn get(&self, key: &str) -> Option<&Value> {
```

Comment:
For the public API this should probably be an [`EvalResult`](https://docs.rs/cedar-policy/latest/cedar_policy/enum.EvalResult.html) instead of a `Value`. I think there's a already a `From` impl for this conversion, though it will mean this function can't return a reference anymore.

Reply:
Thanks, John. I've pushed an update to match these requirements.